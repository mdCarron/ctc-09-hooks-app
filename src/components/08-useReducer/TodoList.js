import React from "react";
import TodoListItem from "./TodoListItem";

const TodoList = ({ todos, handleToggle, handleDelete }) => {
  return (
    <ul className="list-group list-group-flush">
      {todos.map((todo, i) => (
        <TodoListItem
          todo={todo}
          i={i}
          handleToggle={handleToggle}
          handleDelete={handleDelete}
        />
      ))}
    </ul>
  );
};

export default TodoList;
