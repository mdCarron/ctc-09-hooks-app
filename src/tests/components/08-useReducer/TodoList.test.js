import React from "react";
import { shallow } from "enzyme";
import { demoTodos } from "../../fixtures/demoTodos";
import TodoList from "../../../components/08-useReducer/TodoList";

describe("<TodoList />", () => {
  const handleToggle = jest.fn();
  const handleDelete = jest.fn();

  const wrapper = shallow(
    <TodoList
      todos={demoTodos}
      handleToggle={handleToggle}
      handleDelete={handleDelete}
    />
  );

  test("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should have 2 child <TodoListItem />", () => {
    expect(wrapper.find("TodoListItem").length).toBe(demoTodos.length);

    expect(wrapper.find("TodoListItem").at(0).prop("handleToggle")).toEqual(
      expect.any(Function)
    );
  });
});
