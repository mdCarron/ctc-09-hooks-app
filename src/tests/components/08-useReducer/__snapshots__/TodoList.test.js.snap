// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> should match snapshot 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={0}
    todo={
      Object {
        "desc": "Aprender React",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={1}
    todo={
      Object {
        "desc": "Aprender MongoDB",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
